### ISSUE TITLE
Self reports: Second delivery

### ISSUE THREAD

**Issue by UO283615** [Arquisoft/wiq_en2b#107]
Please comment on this issue with your self reports. Remember to vote on each other's reports and to follow the template that we used in the previous one.

**Comment by gony02** [comment]
For this delivery I worked on:
### Pull requests 🪢
#### Related with the Documentation:
- #54
- #99 

#### Related with the webapp:
- #64
- #65
- #79
- #90
- #94 
- #102

### Issues ✍️
I have collaborated in the following issues:
- #51 
- #53 
- #59 
- #60 
- #67 
- #69 
- #74 
- #80 
- #95

### Wiki 📖
Moreover, I have written several pages on the Wiki and I have actively engaged in various meetings. I have not participated in the meetings focused in the backend.
- [Home](https://github.com/Arquisoft/wiq_en2b/wiki)
- [20/02/2024](https://github.com/Arquisoft/wiq_en2b/wiki/Minutes-for-the-20-02-2024-laboratory-session)
- [23/02/2024](https://github.com/Arquisoft/wiq_en2b/wiki/Minutes-for-the-23-02-2024-frontend-meeting)
- [27/02/2024](https://github.com/Arquisoft/wiq_en2b/wiki/Minutes-for-the-27-02-2024-laboratory-session)
- [03/03/2024](https://github.com/Arquisoft/wiq_en2b/wiki/Minutes-for-the-03-03-2024-meeting)
- [05/03/2024](https://github.com/Arquisoft/wiq_en2b/wiki/Minutes-for-the-05-03-2024-laboratory-session)
- [08/03/2024 frontend](https://github.com/Arquisoft/wiq_en2b/wiki/Minutes-for-the-08-03-2024-frontend-meeting)
- [10/03/2024](https://github.com/Arquisoft/wiq_en2b/wiki/Minutes-for-the-10-03-2024-meeting)

### Other 👨‍💻
For this second delivery, I have designed all the wireframes of the application that I have included in the [documentation](https://arquisoft.github.io/wiq_en2b/) (see Section 2: Integration constraints). Then I made the login and logout views in [React](https://react.dev/) with [Chakra UI](https://chakra-ui.com/) with @sergiorodriguezgarcia doing pair programming. Then I made the Dashboard, Rules and Game views. Additionally I have communicated the Game and Rules views with the backend. Finally, I have done unit tests of different views of the webapp trying to have at least 80% coverage in SonarCloud.

**Comment by sergiorodriguezgarcia** [comment]
For this delivery I worked on:
### Pull requests 🪢
#### Related with the webapp:
- #64
- #65
- #84 
- #94 
- #102
- #108 

### Issues ✍️
I have collaborated in the following issues:
- #51 
- #53 
- #59 
- #60 
- #67 
- #69 
- #74 
- #80 
- #95

### Other 👨‍💻
I have made the Login and Logout views in [React](https://react.dev/) with [Chakra UI](https://chakra-ui.com/) with @gony02 doing pair programming. Then, I made some fixes in the Dashboard and game views and additionally, I have communicated the Signup view with the backend and implemented the Logout logic. Finally, I have done unit tests of different views of the webapp trying to have at least 80% coverage in SonarCloud.

**Comment by sergioqfeg1** [comment]
For this delivery I worked on:

### Pull requests 🪢
#### Related with the API:
- #57
- #104
- #89
- #86
- #71 

### Issues ✍️
I have collaborated in the following issues:
- #73 
- #69
- #86
- #66
- #57 
- #51 
- #53 

### Other 👨‍💻
For this seciond delivery my job as part of the backend team, was to develop the controllers, services and repositories for the Springboot API along side @Toto-hitori. We had several pair programming sessions developing the end points and the logic of saving/getting questions and answers. We also developed several tests for the endpoints and the services.

**Comment by uo283928** [comment]
For this delivery I worked on:
# Pull requests 🪢
- #54
- #55 
- #71 
- #85 
- #97 
- #102 
- #103 
- #110 
- #111 
- #112 

# Issues ✍️
I have collaborated in the following issues:
- #53 
- #81 

# Other 👨‍💻
During this sprint, I teamed with @UO283615 to implement the QuestionGenerator module, which has two parts. The first one, querying questions from WikiData QS and the second one, processing and storing the responses into the DB.
The first week we did research and discussions about how to implement it so no code was created. On the second week, Villanueva had more spare time than me so he implemented the module and in the third week I did code clean ups and documented the code so other members of the team can create their own questions easily. 
I also supported @Toto-hitori with the deployment and bug solving from frontend and backend.

**Comment by jjgancfer** [comment]
For this delivery I have worked on:
# Pull requests 🪢
## Reviewed pull requests
- #54
- #62
- #64 
- #65
- #68 
- #70 
- #75 
- #76 
- #79 
- #84 
- #85 
- #87 
- #90 

## Created pull requests

- #55 
- #56 
- #64 
- #65
- #84 

# Issues ✍️
I have collaborated in the following issues:

- #6 
- #51
- #63
- #66 
- #67 
- #69 
- #78 

# Other 👨‍💻
In this sprint, I started developing a new login page before passing that task to @gony02. With him and @sergiorodriguezgarcia, we agreed to discard to the layout initially written in the previous sprint in favour of another. We also teamed up to introduce initial communication between backend and frontend.

Finally, these last days I have included several architectural decisions in the wiki.

**Comment by UO283615** [comment]
For this delivery I worked on:
### Pull requests 🪢
#### Related with the Documentation:
- #98

#### Related with the question generator:
- #71 
- #97 

### Issues ✍️
I have collaborated in the following issues:
- #63 
- #81 
- #53 
- #11 
- #12

### Wiki 📖
Moreover, I have written a page on the Wiki and I have actively engaged in various meetings. I have not participated in the meetings focused in the frontend.
- [Home](https://github.com/Arquisoft/wiq_en2b/wiki)

### Other 👨‍💻
To sum up I focused on the question generator. Specifically, I learnt how to communicate with Wikidata from Java and how to connect with the DB. Also I helped on the design of the data model.

**Comment by Toto-hitori** [comment]
For this delivery I worked on:
### Pull requests 🪢
#### Related with the Documentation:
- #36 

#### PRs:
- #62
- #68 
- #70 
- #71 
- #75 
- #76 
- #82 
- #85
- #86
- #87 
- #89 
- #102 
- #104

### Issues ✍️
I have collaborated in the following issues:
- 

### Wiki 📖
Moreover, I have written several pages on the Wiki and I have actively engaged in various meetings. I have not participated in the meetings focused in the frontend.
- [Home](https://github.com/Arquisoft/wiq_en2b/wiki)
- [20/02/2024](https://github.com/Arquisoft/wiq_en2b/wiki/Minutes-for-the-20-02-2024-laboratory-session)
- [27/02/2024](https://github.com/Arquisoft/wiq_en2b/wiki/Minutes-for-the-27-02-2024-laboratory-session)
- [03/03/2024](https://github.com/Arquisoft/wiq_en2b/wiki/Minutes-for-the-03-03-2024-meeting)
- [05/03/2024](https://github.com/Arquisoft/wiq_en2b/wiki/Minutes-for-the-05-03-2024-laboratory-session)
- [10/03/2024](https://github.com/Arquisoft/wiq_en2b/wiki/Minutes-for-the-10-03-2024-meeting)

### Other 👨‍💻
For this second delivery, I have lost my sanity trying to deploy the api and the webapp, having spent multiple hours deploying and looking into github actions. yay :)

**Comment by uo283928** [comment]
Closed as second delivery review has already been done.
